src/overlays/actors/ovl_Bg_Spot05_Soko/z_bg_spot05_soko.c

```diff
@@ -21,17 +32,77 @@ const ActorInit Bg_Spot05_Soko_InitVars = {
     (ActorFunc)BgSpot05Soko_Update,
     (ActorFunc)BgSpot05Soko_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot05_Soko/BgSpot05Soko_Init.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot05_Soko/BgSpot05Soko_Destroy.s")
+static InitChainEntry initChain[] = {
+    ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_STOP),
+};
+
+static Gfx* dListTbl[] = {
+    0x06000840,
+    0x06001190,
+};
+
+static void BgSpot05Soko_Init(BgSpot05Soko* this, GlobalContext* globalCtx) {
+
+    Actor* thisx = &this->dyna.actor;
+    u32 sp24[2];
+    sp24[0] = 0;
+    Actor_ProcessInitChain(thisx, initChain);
+    this->switchFlag = (thisx->params >> 8) & 0xFF;
+    thisx->params &= 0xFF;
+    DynaPolyInfo_SetActorMove(thisx, DPM_UNK);
+    if (thisx->params == 0) {
+        DynaPolyInfo_Alloc(&D_06000918, &sp24);
+        if (LINK_IS_ADULT) {
+            Actor_Kill(thisx);
+        } else {
+            this->actionFunc = &func_808AE5A8;
+        }
+    } else {
+        DynaPolyInfo_Alloc(&D_060012C0, &sp24);
+        if (Flags_GetSwitch(globalCtx, this->switchFlag) != 0) {
+            Actor_Kill(thisx);
+        } else {
+            this->actionFunc = &func_808AE5B4;
```

Comment:
you dont need & for function pointers or arrays. theres a couple others in this file. other than that and romans comments, looks good